lib/msprime.c

```diff
@@ -2490,49 +2537,104 @@ msp_run_dtwf(msp_t *self, double max_time, unsigned long max_events)
     unsigned long events = 0;
     int mig_source_pop, mig_dest_pop;
     sampling_event_t *se;
-    double mu;
-    uint32_t j, k, i, num_migrations, n;
+    uint32_t j, k, i, N, num_migs, max_migs;
+    unsigned int *n;
```

Comment:
Minor thing here, but need to make sure that the pointers for stuff that's malloced are initialised to NULL and that the return value is checked:
```c
     unsigned int *n = malloc(self->num_populations * sizeof(int));
￼     double *mig_tmp = malloc(self->num_populations * sizeof(double));￼
￼     double sum;
￼     avl_tree_t *node_trees = NULL;
￼     avl_tree_t *nodes;
￼
     if (n == NULL || mig_tmp == NULL) {
        ret  = MSP_ERR_NO_MEMORY;
        goto out;
    }
```
The reason for making sure that pointers (or at least pointers to stuff we eventually malloc/free) are always initialised to NULL is that if an error occurs before malloc is called and we ``goto out`` then we'll try to free some arbitrary memory, and things will explode. This couldn't actually have happened here, but it's better to get into the habit of doing this stuff automatically.


￼

Reply:
Ahh yes, thanks!